demos/demo_spd.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+SPD-Reader demo
+
+Copyright (C) 2020
+               by     Ivo Ihrke
+               
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+
+import sys
+
+import pyrateoptics
+from pyrateoptics import build_rotationally_symmetric_optical_system
+from pyrateoptics import draw
+from pyrateoptics import raytrace
+
+from pyrateoptics.sampling2d.raster import RectGrid
+from pyrateoptics.raytracer.ray import RayBundle
+from pyrateoptics.raytracer.analysis.ray_analysis import RayBundleAnalysis
+from pyrateoptics.raytracer.material.material_glasscat import CatalogMaterial
+from pyrateoptics.raytracer.globalconstants import canonical_ex, canonical_ey
+from pyrateoptics.raytracer.localcoordinates import LocalCoordinates
+from pyrateoptics.raytracer.analysis.ray_analysis import RayBundleAnalysis, RayPathAnalysis
+from pyrateoptics.raytracer.io.spd import SPDParser;
+
+
+from matplotlib import pyplot as plt;
+
+
+#ATTENTION: need to clone first !
+#> git clone https://github.com/polyanskiy/refractiveindex.info-database.git
+db_path = "refractiveindex.info-database/database"
+
+#this way, we may look for names
+gcat = pyrateoptics.GlassCatalog(db_path)
+#print( gcat.find_pages_with_long_name("BK7") )
+
+if gcat.get_shelves() == []:
+    print("""Clone the glass data base first! 
```

Comment:
We have a separate submodule in our repo which is linked to `refractive-index.info` (`pyrateoptics/refractive-index.info`). Maybe it would be better to use a relative path to there and complain about a correct `git submodule update --init --recursive` to be performed in order to use it. At least for the demos this is reasonable since they are part of the repo. The other demos use simplify `refractive-index.info/database`.

Reply:
Ok, I did the glass catalog change. 

For the code style, I am sorry, I do not have the time to change all the issues brought up by the linter. Maybe some other volunteer could approach this later ? I can also keep my own fork in case this is too ugly for you.